### PR TITLE
Improve inbox activity type handling and tests

### PR DESCRIPTION
**Pull request by pfefferle**
Replaces strtolower with camel_to_snake_case for activity and object type normalization in the inbox handler. Updates tests to cover additional cases, including QuoteRequest activities and handling of non-array objects, ensuring more robust and flexible activity type processing.

## Proposed changes:
- Replaces `strtolower` with `camel_to_snake_case` for consistent type normalization
- Adds handling for non-array objects in inbox processing
- Expands test coverage with QuoteRequest activities and additional edge cases

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:

* See unittests

### Changelog entry

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [X] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [X] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Fixed activity processing to handle QuoteRequest and other edge cases more reliably.

</details>

### CODE REVIEW

File: includes/handler/class-inbox.php [hunk 1]
```diff
@@ -56,9 +56,11 @@ public static function handle_inbox_requests( $data, $user_id, $type, $activity
 			 * @param array $object_types The object types to persist in the inbox.
 			 */
 			$object_types = \apply_filters( 'activitypub_persist_inbox_object_types', Base_Object::TYPES );
-			$object_types = \array_map( 'strtolower', $object_types );
+			$object_types = \array_map( 'Activitypub\camel_to_snake_case', $object_types );
 
-			if ( isset( $data['object']['type'] ) && ! \in_array( \strtolower( $data['object']['type'] ), $object_types, true ) ) {
+			if ( ! is_array( $data['object'] ) ) {
+				$success = true;
```
Comment (obenland) [inline includes/handler/class-inbox.php]: At this point `$success` is already true and `$object_types` are unused. Maybe line 52 should be 
```php
// A comment that explains what's being tested for.
if ( $success && is_array( $data['object'] ) ) {
```

**Review by obenland — APPROVED** [review]
